faststream/prometheus/middleware.py

```diff
@@ -0,0 +1,256 @@
+import time
+from typing import TYPE_CHECKING, Any, Callable, Optional
+
+from prometheus_client import Counter, Gauge, Histogram
+
+from faststream import BaseMiddleware
+from faststream.prometheus.consts import (
+    PROCESSING_STATUS_BY_ACK_STATUS,
+    PROCESSING_STATUS_BY_HANDLER_EXCEPTION_MAP,
+)
+from faststream.prometheus.provider import MetricsSettingsProvider
+from faststream.prometheus.types import ProcessingStatus, PublishingStatus
+
+if TYPE_CHECKING:
+    from prometheus_client import CollectorRegistry
+
+    from faststream.broker.message import StreamMessage
+    from faststream.types import AsyncFunc, AsyncFuncAny
+
+
+class _MetricsContainer:
+    __slots__ = (
+        "received_messages",
+        "received_messages_size",
+        "received_messages_processing_time",
+        "received_messages_in_process",
+        "received_processed_messages",
+        "messages_processing_exceptions",
+        "published_messages",
+        "messages_publishing_time",
+        "messages_publishing_exceptions",
+    )
+
+    def __init__(self, registry: "CollectorRegistry"):
+        self.received_messages = Counter(
+            name="received_messages",
+            documentation="Received messages",
+            labelnames=["broker", "handler"],
+            registry=registry,
+        )
+        self.received_messages_size = Histogram(
+            name="received_messages_size",
+            documentation="Received messages size (in bytes)",
+            labelnames=["broker", "handler"],
+            registry=registry,
+            buckets=[
+                pow(2, x) for x in range(31)
```

Comment:
Why are there such buckets?

Reply:
changed the default buckets to those that we discussed with you

also added the received_messages_size_buckets parameter so that the user can specify his own buckets for message sizes